includes/collection/class-outbox.php

```diff
@@ -25,41 +24,34 @@ class Outbox {
 	/**
 	 * Add an Item to the outbox.
 	 *
-	 * @param \Activitypub\Activity\Base_Object $activity_object    The object of the activity that will be added to the outbox.
-	 * @param string                            $activity_type      The activity type.
-	 * @param int                               $user_id            The real or imaginary user ID of the actor that published the activity that will be added to the outbox.
-	 * @param string                            $content_visibility Optional. The visibility of the content. Default: `ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC`. See `constants.php` for possible values: `ACTIVITYPUB_CONTENT_VISIBILITY_*`.
+	 * @param Activity $activity   Full Activity object that will be added to the outbox.
+	 * @param int      $user_id    The real or imaginary user ID of the actor that published the activity that will be added to the outbox.
+	 * @param string   $visibility Optional. The visibility of the content. Default: `ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC`. See `constants.php` for possible values: `ACTIVITYPUB_CONTENT_VISIBILITY_*`.
 	 *
 	 * @return false|int|\WP_Error The added item or an error.
 	 */
-	public static function add( $activity_object, $activity_type, $user_id, $content_visibility = ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC ) {
-		$actor_type            = Actors::get_type_by_id( $user_id );
-		$title                 = $activity_object->get_name() ?? $activity_object->get_content();
-		$activitypub_object_id = $activity_object->get_id();
-
-		if ( ! $title && is_activity( $activity_object ) && $activity_object->get_object() instanceof \Activitypub\Activity\Base_Object ) {
-			$title                 = $activity_object->get_object()->get_name() ?? $activity_object->get_object()->get_content();
-			$activitypub_object_id = $activity_object->get_object()->get_id();
-		}
+	public static function add( Activity $activity, $user_id, $visibility = ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC ) {
+		$actor_type = Actors::get_type_by_id( $user_id );
+		$object_id  = is_string( $activity->get_object() ) ? $activity->get_object() : $activity->get_object()->get_id();
```

Comment:
```suggestion
		$object_id = false;
		if ( $activity->get_object() ) {
			$object_id  = is_string( $activity->get_object() ) ? $activity->get_object() : $activity->get_object()->get_id();
		}
```

An `object` is not required!

Reply:
Any repercussions from not having an object id saved?